core/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -362,6 +365,8 @@ private QueryFetchSearchResult executeFetchPhase(SearchContext context, SearchOp
 
     public ScrollQuerySearchResult executeQueryPhase(InternalScrollSearchRequest request, SearchTask task) {
         final SearchContext context = findContext(request.id(), request);
+        if (context == null)
+            return new ScrollQuerySearchResult();
```

Comment:
Hi @javanna, i currently struggling with what we should return here. If we return `new ScrollQuerySearchResult()` i get this error for a scroll request
```
 "shard" : -1,
        "index" : null,
        "reason" : {
          "type" : "response_handler_failure_transport_exception",
          "reason" : "java.lang.NullPointerException",
          "caused_by" : {
            "type" : "null_pointer_exception",
            "reason" : null
```
I dont't find the point where we should return an empty result set.